assets/src/blocks/Counter/CounterFrontend.js

```diff
@@ -0,0 +1,123 @@
+import { Component, Fragment } from '@wordpress/element';
+
+export class CounterFrontend extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      remaining: 0,
+      completed: 0
+    };
+
+    this.calculateRemaining = this.calculateRemaining.bind(this);
+    this.getCounterText = this.getCounterText.bind(this);
+  }
+
+  componentDidMount() {
+    // Calculate completed and remaining values depending on props
+    this.calculateRemaining();
+  }
+
+  componentDidUpdate({ target: prevTarget, completed: prevCompleted, completed_api: prevCompletedApi }) {
+    // Update completed and remaining values depending on props
+    const { target, completed, completed_api } = this.props;
+    if (target !== prevTarget || completed !== prevCompleted || completed_api !== prevCompletedApi) {
+      this.calculateRemaining();
+    }
+  }
+
+  calculateRemaining() {
+    const { completed_api } = this.props;
+    const target = this.props.target || 0;
+    let completed = this.props.completed || 0;
+    let remaining = 0;
+    if (completed_api) {
+      fetch(completed_api)
+        .then(response => response.json())
+        .then(({ unique_count }) => {
+          if (unique_count) {
+            completed = unique_count;
+            this.setState({
+              completed,
+              remaining: target - completed
+            });
+          }
+        });
+    } else if (target > 0 || completed > 0) {
```

Comment:
is `||` voluntary or did you mean `&&` ?

Reply:
It was voluntary 😬 But considering your previous remarks I'll probably change this part!

Reply:
I ended up removing this check, I think it's no longer necessary with the `Math.max`